pkg/volume/image/image.go

```diff
@@ -0,0 +1,80 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package image
+
+import (
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/kubernetes/pkg/volume"
+)
+
+// imagePlugin is the image volume plugin which acts as a stub to provide the
+// functionality the volume manager expects. The real volume source
+// implementation is part of the kubelet code and gated by the Kubernetes
+// feature "ImageVolume"
+// See: https://kep.k8s.io/4639
+type imagePlugin struct {
+	spec *volume.Spec
+	volume.MetricsNil
+}
+
+var _ volume.VolumePlugin = &imagePlugin{}
```

Comment:
nit: I know this implementation is mostly a stub, but can you add declaration like https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/local/local.go#L60 to indicate interfaces that this plugin implements?

Reply:
Hm, my goal was to go with the most minimal approach to keep the code surface low. We also do not support any resizing so I'm wondering if there is any other specific interface you're looking for?

Reply:
Well, apart from `VolumePlugin` interface - we clearly implement `Mounter`, `Umounter` and `Volume` interfaces.

Reply:
Makes sense, added all three of them :+1: